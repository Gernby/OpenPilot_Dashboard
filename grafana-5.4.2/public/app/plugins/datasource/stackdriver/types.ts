export enum MetricFindQueryTypes {
  Services = 'services',
  MetricTypes = 'metricTypes',
  LabelKeys = 'labelKeys',
  LabelValues = 'labelValues',
  ResourceTypes = 'resourceTypes',
  Aggregations = 'aggregations',
  Aligners = 'aligners',
  AlignmentPeriods = 'alignmentPeriods',
}

export interface VariableQueryData {
  selectedQueryType: string;
  metricDescriptors: any[];
  selectedService: string;
  selectedMetricType: string;
  labels: string[];
  labelKey: string;
  metricTypes: Array<{ value: string; name: string }>;
  services: Array<{ value: string; name: string }>;
}
