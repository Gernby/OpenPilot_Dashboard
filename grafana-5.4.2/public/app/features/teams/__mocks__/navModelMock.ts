export const getMockNavModel = (pageName: string) => {
  return {
    node: {
      active: false,
      icon: 'gicon gicon-team',
      id: `team-${pageName}-2`,
      text: `${pageName}`,
      url: 'org/teams/edit/2/members',
      parentItem: {
        img: '/avatar/b5695b61c91d13e7fa2fe71cfb95de9b',
        id: 'team-2',
        subTitle: 'Manage members & settings',
        url: '',
        text: 'test1',
        breadcrumbs: [{ title: 'Teams', url: 'org/teams' }],
        children: [
          {
            active: false,
            icon: 'gicon gicon-team',
            id: 'team-members-2',
            text: 'Members',
            url: 'org/teams/edit/2/members',
          },
          {
            active: false,
            icon: 'fa fa-fw fa-sliders',
            id: 'team-settings-2',
            text: 'Settings',
            url: 'org/teams/edit/2/settings',
          },
        ],
      },
    },
    main: {
      img: '/avatar/b5695b61c91d13e7fa2fe71cfb95de9b',
      id: 'team-2',
      subTitle: 'Manage members & settings',
      url: '',
      text: 'test1',
      breadcrumbs: [{ title: 'Teams', url: 'org/teams' }],
      children: [
        {
          active: true,
          icon: 'gicon gicon-team',
          id: 'team-members-2',
          text: 'Members',
          url: 'org/teams/edit/2/members',
        },
        {
          active: false,
          icon: 'fa fa-fw fa-sliders',
          id: 'team-settings-2',
          text: 'Settings',
          url: 'org/teams/edit/2/settings',
        },
      ],
    },
  };
};
