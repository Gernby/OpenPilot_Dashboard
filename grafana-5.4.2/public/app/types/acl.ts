export enum OrgRole {
  Viewer = 'Viewer',
  Editor = 'Editor',
  Admin = 'Admin',
}

export interface DashboardAclDTO {
  id?: number;
  dashboardId?: number;
  userId?: number;
  userLogin?: string;
  userEmail?: string;
  teamId?: number;
  team?: string;
  permission?: PermissionLevel;
  role?: OrgRole;
  icon?: string;
  inherited?: boolean;
}

export interface DashboardAclUpdateDTO {
  userId: number;
  teamId: number;
  role: OrgRole;
  permission: PermissionLevel;
}

export interface DashboardAcl {
  id?: number;
  dashboardId?: number;
  userId?: number;
  userLogin?: string;
  userEmail?: string;
  teamId?: number;
  team?: string;
  permission?: PermissionLevel;
  role?: OrgRole;
  icon?: string;
  name?: string;
  inherited?: boolean;
  sortRank?: number;
}

export interface DashboardPermissionInfo {
  value: PermissionLevel;
  label: string;
  description: string;
}

export interface NewDashboardAclItem {
  teamId: number;
  userId: number;
  role?: OrgRole;
  permission: PermissionLevel;
  type: AclTarget;
}

export enum PermissionLevel {
  View = 1,
  Edit = 2,
  Admin = 4,
}

export enum DataSourcePermissionLevel {
  Query = 1,
  Admin = 2,
}

export enum AclTarget {
  Team = 'Team',
  User = 'User',
  Viewer = 'Viewer',
  Editor = 'Editor',
}

export interface AclTargetInfo {
  value: AclTarget;
  text: string;
}

export const dataSourceAclLevels = [
  { value: DataSourcePermissionLevel.Query, label: 'Query', description: 'Can query data source.' },
];

export const dashboardAclTargets: AclTargetInfo[] = [
  { value: AclTarget.Team, text: 'Team' },
  { value: AclTarget.User, text: 'User' },
  { value: AclTarget.Viewer, text: 'Everyone With Viewer Role' },
  { value: AclTarget.Editor, text: 'Everyone With Editor Role' },
];

export const dashboardPermissionLevels: DashboardPermissionInfo[] = [
  { value: PermissionLevel.View, label: 'View', description: 'Can view dashboards.' },
  { value: PermissionLevel.Edit, label: 'Edit', description: 'Can add, edit and delete dashboards.' },
  {
    value: PermissionLevel.Admin,
    label: 'Admin',
    description: 'Can add/remove permissions and can add, edit and delete dashboards.',
  },
];
