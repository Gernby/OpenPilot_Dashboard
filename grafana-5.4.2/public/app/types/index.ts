import { Team, TeamsState, TeamState, TeamGroup, TeamMember } from './teams';
import { AlertRuleDTO, AlertRule, AlertRulesState } from './alerting';
import { LocationState, LocationUpdate, UrlQueryMap, UrlQueryValue } from './location';
import { NavModel, NavModelItem, NavIndex } from './navModel';
import { FolderDTO, FolderState, FolderInfo } from './folders';
import { DashboardState } from './dashboard';
import { DashboardAcl, OrgRole, PermissionLevel } from './acl';
import { ApiKey, ApiKeysState, NewApiKey } from './apiKeys';
import { Invitee, OrgUser, User, UsersState, UserState } from './user';
import { DataSource, DataSourcesState } from './datasources';
import {
  TimeRange,
  LoadingState,
  TimeSeries,
  TimeSeriesVM,
  TimeSeriesVMs,
  TimeSeriesStats,
  NullValueMode,
  DataQuery,
  DataQueryResponse,
  DataQueryOptions,
} from './series';
import { PanelProps, PanelOptionsProps } from './panel';
import { PluginDashboard, PluginMeta, Plugin, PluginsState } from './plugins';
import { Organization, OrganizationState } from './organization';
import {
  AppNotification,
  AppNotificationSeverity,
  AppNotificationsState,
  AppNotificationTimeout,
} from './appNotifications';
import { DashboardSearchHit } from './search';

export {
  Team,
  TeamsState,
  TeamState,
  TeamGroup,
  TeamMember,
  AlertRuleDTO,
  AlertRule,
  AlertRulesState,
  LocationState,
  LocationUpdate,
  NavModel,
  NavModelItem,
  NavIndex,
  UrlQueryMap,
  UrlQueryValue,
  FolderDTO,
  FolderState,
  FolderInfo,
  DashboardState,
  DashboardAcl,
  OrgRole,
  PermissionLevel,
  DataSource,
  PluginMeta,
  ApiKey,
  ApiKeysState,
  NewApiKey,
  Plugin,
  PluginsState,
  DataSourcesState,
  Invitee,
  OrgUser,
  User,
  UsersState,
  TimeRange,
  LoadingState,
  PanelProps,
  PanelOptionsProps,
  TimeSeries,
  TimeSeriesVM,
  TimeSeriesVMs,
  NullValueMode,
  TimeSeriesStats,
  DataQuery,
  DataQueryResponse,
  DataQueryOptions,
  PluginDashboard,
  Organization,
  OrganizationState,
  AppNotification,
  AppNotificationsState,
  AppNotificationSeverity,
  AppNotificationTimeout,
  DashboardSearchHit,
  UserState,
};

export interface StoreState {
  navIndex: NavIndex;
  location: LocationState;
  alertRules: AlertRulesState;
  teams: TeamsState;
  team: TeamState;
  folder: FolderState;
  dashboard: DashboardState;
  dataSources: DataSourcesState;
  users: UsersState;
  organization: OrganizationState;
  appNotifications: AppNotificationsState;
  user: UserState;
}
