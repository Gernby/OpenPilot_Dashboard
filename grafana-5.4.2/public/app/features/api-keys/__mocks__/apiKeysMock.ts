import { ApiKey, OrgRole } from 'app/types';

export const getMultipleMockKeys = (numberOfKeys: number): ApiKey[] => {
  const keys: ApiKey[] = [];
  for (let i = 1; i <= numberOfKeys; i++) {
    keys.push({
      id: i,
      name: `test-${i}`,
      role: OrgRole.Viewer,
    });
  }

  return keys;
};

export const getMockKey = (): ApiKey => {
  return {
    id: 1,
    name: 'test',
    role: OrgRole.Admin,
  };
};
