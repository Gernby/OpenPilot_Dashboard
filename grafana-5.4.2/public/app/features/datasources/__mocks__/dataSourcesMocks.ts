import { DataSource } from 'app/types';

export const getMockDataSources = (amount: number): DataSource[] => {
  const dataSources = [];

  for (let i = 0; i <= amount; i++) {
    dataSources.push({
      access: '',
      basicAuth: false,
      database: `database-${i}`,
      id: i,
      isDefault: false,
      jsonData: { authType: 'credentials', defaultRegion: 'eu-west-2' },
      name: `dataSource-${i}`,
      orgId: 1,
      password: '',
      readOnly: false,
      type: 'cloudwatch',
      typeLogoUrl: 'public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png',
      url: '',
      user: '',
    });
  }

  return dataSources;
};

export const getMockDataSource = (): DataSource => {
  return {
    access: '',
    basicAuth: false,
    database: '',
    id: 13,
    isDefault: false,
    jsonData: { authType: 'credentials', defaultRegion: 'eu-west-2' },
    name: 'gdev-cloudwatch',
    orgId: 1,
    password: '',
    readOnly: false,
    type: 'cloudwatch',
    typeLogoUrl: 'public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png',
    url: '',
    user: '',
  };
};
