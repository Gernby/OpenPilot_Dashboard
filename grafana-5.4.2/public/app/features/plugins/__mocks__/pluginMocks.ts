import { Plugin } from 'app/types';

export const getMockPlugins = (amount: number): Plugin[] => {
  const plugins = [];

  for (let i = 0; i <= amount; i++) {
    plugins.push({
      defaultNavUrl: 'some/url',
      enabled: false,
      hasUpdate: false,
      id: `${i}`,
      info: {
        author: {
          name: 'Grafana Labs',
          url: 'url/to/GrafanaLabs',
        },
        description: 'pretty decent plugin',
        links: ['one link'],
        logos: { small: 'small/logo', large: 'large/logo' },
        screenshots: `screenshot/${i}`,
        updated: '2018-09-26',
        version: '1',
      },
      latestVersion: `1.${i}`,
      name: `pretty cool plugin-${i}`,
      pinned: false,
      state: '',
      type: '',
    });
  }

  return plugins;
};

export const getMockPlugin = () => {
  return {
    defaultNavUrl: 'some/url',
    enabled: false,
    hasUpdate: false,
    id: '1',
    info: {
      author: {
        name: 'Grafana Labs',
        url: 'url/to/GrafanaLabs',
      },
      description: 'pretty decent plugin',
      links: ['one link'],
      logos: { small: 'small/logo', large: 'large/logo' },
      screenshots: 'screenshot/1',
      updated: '2018-09-26',
      version: '1',
    },
    latestVersion: '1',
    name: 'pretty cool plugin 1',
    pinned: false,
    state: '',
    type: '',
  };
};
