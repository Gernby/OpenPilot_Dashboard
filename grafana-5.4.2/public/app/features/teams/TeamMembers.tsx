import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import SlideDown from 'app/core/components/Animations/SlideDown';
import { UserPicker } from 'app/core/components/Picker/UserPicker';
import DeleteButton from 'app/core/components/DeleteButton/DeleteButton';
import { TagBadge } from 'app/core/components/TagFilter/TagBadge';
import { TeamMember, User } from 'app/types';
import { loadTeamMembers, addTeamMember, removeTeamMember, setSearchMemberQuery } from './state/actions';
import { getSearchMemberQuery, getTeamMembers } from './state/selectors';

export interface Props {
  members: TeamMember[];
  searchMemberQuery: string;
  loadTeamMembers: typeof loadTeamMembers;
  addTeamMember: typeof addTeamMember;
  removeTeamMember: typeof removeTeamMember;
  setSearchMemberQuery: typeof setSearchMemberQuery;
  syncEnabled: boolean;
}

export interface State {
  isAdding: boolean;
  newTeamMember?: User;
}

export class TeamMembers extends PureComponent<Props, State> {
  constructor(props) {
    super(props);
    this.state = { isAdding: false, newTeamMember: null };
  }

  componentDidMount() {
    this.props.loadTeamMembers();
  }

  onSearchQueryChange = event => {
    this.props.setSearchMemberQuery(event.target.value);
  };

  onRemoveMember(member: TeamMember) {
    this.props.removeTeamMember(member.userId);
  }

  onToggleAdding = () => {
    this.setState({ isAdding: !this.state.isAdding });
  };

  onUserSelected = (user: User) => {
    this.setState({ newTeamMember: user });
  };

  onAddUserToTeam = async () => {
    this.props.addTeamMember(this.state.newTeamMember.id);
    this.setState({ newTeamMember: null });
  };

  renderLabels(labels: string[]) {
    if (!labels) {
      return <td />;
    }

    return (
      <td>
        {labels.map(label => <TagBadge key={label} label={label} removeIcon={false} count={0} onClick={() => {}} />)}
      </td>
    );
  }

  renderMember(member: TeamMember, syncEnabled: boolean) {
    return (
      <tr key={member.userId}>
        <td className="width-4 text-center">
          <img className="filter-table__avatar" src={member.avatarUrl} />
        </td>
        <td>{member.login}</td>
        <td>{member.email}</td>
        {syncEnabled && this.renderLabels(member.labels)}
        <td className="text-right">
          <DeleteButton onConfirmDelete={() => this.onRemoveMember(member)} />
        </td>
      </tr>
    );
  }

  render() {
    const { isAdding } = this.state;
    const { searchMemberQuery, members, syncEnabled } = this.props;
    return (
      <div>
        <div className="page-action-bar">
          <div className="gf-form gf-form--grow">
            <label className="gf-form--has-input-icon gf-form--grow">
              <input
                type="text"
                className="gf-form-input"
                placeholder="Search members"
                value={searchMemberQuery}
                onChange={this.onSearchQueryChange}
              />
              <i className="gf-form-input-icon fa fa-search" />
            </label>
          </div>

          <div className="page-action-bar__spacer" />

          <button className="btn btn-success pull-right" onClick={this.onToggleAdding} disabled={isAdding}>
            <i className="fa fa-plus" /> Add a member
          </button>
        </div>

        <SlideDown in={isAdding}>
          <div className="cta-form">
            <button className="cta-form__close btn btn-transparent" onClick={this.onToggleAdding}>
              <i className="fa fa-close" />
            </button>
            <h5>Add Team Member</h5>
            <div className="gf-form-inline">
              <UserPicker onSelected={this.onUserSelected} className="min-width-30" />
              {this.state.newTeamMember && (
                <button className="btn btn-success gf-form-btn" type="submit" onClick={this.onAddUserToTeam}>
                  Add to team
                </button>
              )}
            </div>
          </div>
        </SlideDown>

        <div className="admin-list-table">
          <table className="filter-table filter-table--hover form-inline">
            <thead>
              <tr>
                <th />
                <th>Name</th>
                <th>Email</th>
                {syncEnabled && <th />}
                <th style={{ width: '1%' }} />
              </tr>
            </thead>
            <tbody>{members && members.map(member => this.renderMember(member, syncEnabled))}</tbody>
          </table>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    members: getTeamMembers(state.team),
    searchMemberQuery: getSearchMemberQuery(state.team),
  };
}

const mapDispatchToProps = {
  loadTeamMembers,
  addTeamMember,
  removeTeamMember,
  setSearchMemberQuery,
};

export default connect(mapStateToProps, mapDispatchToProps)(TeamMembers);
