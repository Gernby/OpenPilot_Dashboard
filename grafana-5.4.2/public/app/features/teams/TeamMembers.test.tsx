import React from 'react';
import { shallow } from 'enzyme';
import { TeamMembers, Props, State } from './TeamMembers';
import { TeamMember } from '../../types';
import { getMockTeamMember, getMockTeamMembers } from './__mocks__/teamMocks';

const setup = (propOverrides?: object) => {
  const props: Props = {
    members: [] as TeamMember[],
    searchMemberQuery: '',
    setSearchMemberQuery: jest.fn(),
    loadTeamMembers: jest.fn(),
    addTeamMember: jest.fn(),
    removeTeamMember: jest.fn(),
    syncEnabled: false,
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<TeamMembers {...props} />);
  const instance = wrapper.instance() as TeamMembers;

  return {
    wrapper,
    instance,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render team members', () => {
    const { wrapper } = setup({
      members: getMockTeamMembers(5),
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render team members when sync enabled', () => {
    const { wrapper } = setup({
      members: getMockTeamMembers(5),
      syncEnabled: true,
    });

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Functions', () => {
  describe('on search member query change', () => {
    it('it should call setSearchMemberQuery', () => {
      const { instance } = setup();
      const mockEvent = { target: { value: 'member' } };

      instance.onSearchQueryChange(mockEvent);

      expect(instance.props.setSearchMemberQuery).toHaveBeenCalledWith('member');
    });
  });

  describe('on remove member', () => {
    const { instance } = setup();
    const mockTeamMember = getMockTeamMember();

    instance.onRemoveMember(mockTeamMember);

    expect(instance.props.removeTeamMember).toHaveBeenCalledWith(1);
  });

  describe('on add user to team', () => {
    const { wrapper, instance } = setup();
    const state = wrapper.state() as State;

    state.newTeamMember = {
      id: 1,
      label: '',
      avatarUrl: '',
      login: '',
    };

    instance.onAddUserToTeam();

    expect(instance.props.addTeamMember).toHaveBeenCalledWith(1);
  });
});
