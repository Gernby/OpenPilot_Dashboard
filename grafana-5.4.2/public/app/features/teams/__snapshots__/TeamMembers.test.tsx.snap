// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div>
  <div
    className="page-action-bar"
  >
    <div
      className="gf-form gf-form--grow"
    >
      <label
        className="gf-form--has-input-icon gf-form--grow"
      >
        <input
          className="gf-form-input"
          onChange={[Function]}
          placeholder="Search members"
          type="text"
          value=""
        />
        <i
          className="gf-form-input-icon fa fa-search"
        />
      </label>
    </div>
    <div
      className="page-action-bar__spacer"
    />
    <button
      className="btn btn-success pull-right"
      disabled={false}
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
       Add a member
    </button>
  </div>
  <Component
    in={false}
  >
    <div
      className="cta-form"
    >
      <button
        className="cta-form__close btn btn-transparent"
        onClick={[Function]}
      >
        <i
          className="fa fa-close"
        />
      </button>
      <h5>
        Add Team Member
      </h5>
      <div
        className="gf-form-inline"
      >
        <UserPicker
          className="min-width-30"
          onSelected={[Function]}
        />
      </div>
    </div>
  </Component>
  <div
    className="admin-list-table"
  >
    <table
      className="filter-table filter-table--hover form-inline"
    >
      <thead>
        <tr>
          <th />
          <th>
            Name
          </th>
          <th>
            Email
          </th>
          <th
            style={
              Object {
                "width": "1%",
              }
            }
          />
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Render should render team members 1`] = `
<div>
  <div
    className="page-action-bar"
  >
    <div
      className="gf-form gf-form--grow"
    >
      <label
        className="gf-form--has-input-icon gf-form--grow"
      >
        <input
          className="gf-form-input"
          onChange={[Function]}
          placeholder="Search members"
          type="text"
          value=""
        />
        <i
          className="gf-form-input-icon fa fa-search"
        />
      </label>
    </div>
    <div
      className="page-action-bar__spacer"
    />
    <button
      className="btn btn-success pull-right"
      disabled={false}
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
       Add a member
    </button>
  </div>
  <Component
    in={false}
  >
    <div
      className="cta-form"
    >
      <button
        className="cta-form__close btn btn-transparent"
        onClick={[Function]}
      >
        <i
          className="fa fa-close"
        />
      </button>
      <h5>
        Add Team Member
      </h5>
      <div
        className="gf-form-inline"
      >
        <UserPicker
          className="min-width-30"
          onSelected={[Function]}
        />
      </div>
    </div>
  </Component>
  <div
    className="admin-list-table"
  >
    <table
      className="filter-table filter-table--hover form-inline"
    >
      <thead>
        <tr>
          <th />
          <th>
            Name
          </th>
          <th>
            Email
          </th>
          <th
            style={
              Object {
                "width": "1%",
              }
            }
          />
        </tr>
      </thead>
      <tbody>
        <tr
          key="1"
        >
          <td
            className="width-4 text-center"
          >
            <img
              className="filter-table__avatar"
              src="some/url/"
            />
          </td>
          <td>
            testUser-1
          </td>
          <td>
            test@test.com
          </td>
          <td
            className="text-right"
          >
            <DeleteButton
              onConfirmDelete={[Function]}
            />
          </td>
        </tr>
        <tr
          key="2"
        >
          <td
            className="width-4 text-center"
          >
            <img
              className="filter-table__avatar"
              src="some/url/"
            />
          </td>
          <td>
            testUser-2
          </td>
          <td>
            test@test.com
          </td>
          <td
            className="text-right"
          >
            <DeleteButton
              onConfirmDelete={[Function]}
            />
          </td>
        </tr>
        <tr
          key="3"
        >
          <td
            className="width-4 text-center"
          >
            <img
              className="filter-table__avatar"
              src="some/url/"
            />
          </td>
          <td>
            testUser-3
          </td>
          <td>
            test@test.com
          </td>
          <td
            className="text-right"
          >
            <DeleteButton
              onConfirmDelete={[Function]}
            />
          </td>
        </tr>
        <tr
          key="4"
        >
          <td
            className="width-4 text-center"
          >
            <img
              className="filter-table__avatar"
              src="some/url/"
            />
          </td>
          <td>
            testUser-4
          </td>
          <td>
            test@test.com
          </td>
          <td
            className="text-right"
          >
            <DeleteButton
              onConfirmDelete={[Function]}
            />
          </td>
        </tr>
        <tr
          key="5"
        >
          <td
            className="width-4 text-center"
          >
            <img
              className="filter-table__avatar"
              src="some/url/"
            />
          </td>
          <td>
            testUser-5
          </td>
          <td>
            test@test.com
          </td>
          <td
            className="text-right"
          >
            <DeleteButton
              onConfirmDelete={[Function]}
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Render should render team members when sync enabled 1`] = `
<div>
  <div
    className="page-action-bar"
  >
    <div
      className="gf-form gf-form--grow"
    >
      <label
        className="gf-form--has-input-icon gf-form--grow"
      >
        <input
          className="gf-form-input"
          onChange={[Function]}
          placeholder="Search members"
          type="text"
          value=""
        />
        <i
          className="gf-form-input-icon fa fa-search"
        />
      </label>
    </div>
    <div
      className="page-action-bar__spacer"
    />
    <button
      className="btn btn-success pull-right"
      disabled={false}
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
       Add a member
    </button>
  </div>
  <Component
    in={false}
  >
    <div
      className="cta-form"
    >
      <button
        className="cta-form__close btn btn-transparent"
        onClick={[Function]}
      >
        <i
          className="fa fa-close"
        />
      </button>
      <h5>
        Add Team Member
      </h5>
      <div
        className="gf-form-inline"
      >
        <UserPicker
          className="min-width-30"
          onSelected={[Function]}
        />
      </div>
    </div>
  </Component>
  <div
    className="admin-list-table"
  >
    <table
      className="filter-table filter-table--hover form-inline"
    >
      <thead>
        <tr>
          <th />
          <th>
            Name
          </th>
          <th>
            Email
          </th>
          <th />
          <th
            style={
              Object {
                "width": "1%",
              }
            }
          />
        </tr>
      </thead>
      <tbody>
        <tr
          key="1"
        >
          <td
            className="width-4 text-center"
          >
            <img
              className="filter-table__avatar"
              src="some/url/"
            />
          </td>
          <td>
            testUser-1
          </td>
          <td>
            test@test.com
          </td>
          <td>
            <TagBadge
              count={0}
              key="label 1"
              label="label 1"
              onClick={[Function]}
              removeIcon={false}
            />
            <TagBadge
              count={0}
              key="label 2"
              label="label 2"
              onClick={[Function]}
              removeIcon={false}
            />
          </td>
          <td
            className="text-right"
          >
            <DeleteButton
              onConfirmDelete={[Function]}
            />
          </td>
        </tr>
        <tr
          key="2"
        >
          <td
            className="width-4 text-center"
          >
            <img
              className="filter-table__avatar"
              src="some/url/"
            />
          </td>
          <td>
            testUser-2
          </td>
          <td>
            test@test.com
          </td>
          <td>
            <TagBadge
              count={0}
              key="label 1"
              label="label 1"
              onClick={[Function]}
              removeIcon={false}
            />
            <TagBadge
              count={0}
              key="label 2"
              label="label 2"
              onClick={[Function]}
              removeIcon={false}
            />
          </td>
          <td
            className="text-right"
          >
            <DeleteButton
              onConfirmDelete={[Function]}
            />
          </td>
        </tr>
        <tr
          key="3"
        >
          <td
            className="width-4 text-center"
          >
            <img
              className="filter-table__avatar"
              src="some/url/"
            />
          </td>
          <td>
            testUser-3
          </td>
          <td>
            test@test.com
          </td>
          <td>
            <TagBadge
              count={0}
              key="label 1"
              label="label 1"
              onClick={[Function]}
              removeIcon={false}
            />
            <TagBadge
              count={0}
              key="label 2"
              label="label 2"
              onClick={[Function]}
              removeIcon={false}
            />
          </td>
          <td
            className="text-right"
          >
            <DeleteButton
              onConfirmDelete={[Function]}
            />
          </td>
        </tr>
        <tr
          key="4"
        >
          <td
            className="width-4 text-center"
          >
            <img
              className="filter-table__avatar"
              src="some/url/"
            />
          </td>
          <td>
            testUser-4
          </td>
          <td>
            test@test.com
          </td>
          <td>
            <TagBadge
              count={0}
              key="label 1"
              label="label 1"
              onClick={[Function]}
              removeIcon={false}
            />
            <TagBadge
              count={0}
              key="label 2"
              label="label 2"
              onClick={[Function]}
              removeIcon={false}
            />
          </td>
          <td
            className="text-right"
          >
            <DeleteButton
              onConfirmDelete={[Function]}
            />
          </td>
        </tr>
        <tr
          key="5"
        >
          <td
            className="width-4 text-center"
          >
            <img
              className="filter-table__avatar"
              src="some/url/"
            />
          </td>
          <td>
            testUser-5
          </td>
          <td>
            test@test.com
          </td>
          <td>
            <TagBadge
              count={0}
              key="label 1"
              label="label 1"
              onClick={[Function]}
              removeIcon={false}
            />
            <TagBadge
              count={0}
              key="label 2"
              label="label 2"
              onClick={[Function]}
              removeIcon={false}
            />
          </td>
          <td
            className="text-right"
          >
            <DeleteButton
              onConfirmDelete={[Function]}
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
